agent-backend/src/tools/retrievers/self_query.py

```diff
@@ -0,0 +1,23 @@
+from langchain.chains.query_constructor.schema import AttributeInfo
+from langchain.retrievers import SelfQueryRetriever as LC_SelfQueryRetriever
+from langchain_core.language_models import BaseLanguageModel
+from langchain_core.vectorstores import VectorStore
+
+from models.mongo import Tool
+from .base import BaseToolRetriever
+
+
+class SelfQueryRetriever(BaseToolRetriever):
+    def __init__(self, tool: Tool, llm: BaseLanguageModel, vector_store: VectorStore):
+        metadata_field_info = list(
+            map(lambda x: AttributeInfo(**x.model_dump()), tool.retriever_config.metadata_field_info))
+
+        # TODO: Change the `document_contents` param value from `tool.description` to something else
+        self.retriever = LC_SelfQueryRetriever.from_llm(llm=llm,
+                                                        vectorstore=vector_store,
+                                                        document_contents=tool.description,
```

Comment:
This parameter is meant to describe the contents of the document being queried. Should we introduce a new sub-field for this or refer to the datasource's description?

Reply:
tool description is correct